src/timestamp.ts

```diff
@@ -1,20 +1,21 @@
-'use strict';
-
-const Long = require('./long');
+import { Long } from './long';
 
 /**
  * @class
  * @param {number} low  the low (signed) 32 bits of the Timestamp.
  * @param {number} high the high (signed) 32 bits of the Timestamp.
  * @return {Timestamp}
  */
-class Timestamp extends Long {
-  constructor(low, high) {
+export class Timestamp extends Long {
+  constructor(low: Long);
+  constructor(low: number, high: number);
+  constructor(low: number | Long, high?: number) {
     if (Long.isLong(low)) {
       super(low.low, low.high, true);
     } else {
       super(low, high, true);
     }
+    this._bsontype = 'Timestamp';
```

Comment:
There's a crash I get where it says you can't write _bsontype to readonly property. But only on the browser tests, I think this is because the generated bundles make exports from other modules readonly? 🤔 so like Timestamp.prototype is Long, and that's readonly. Not sure, but I'm investigating, threw up a potential solution in the meantime.